### DISCUSSION TITLE
Questions about chargecloud models of SSD

### DISCUSSION THREAD

**Discussion by zql2021** [JuliaPhysics/SolidStateDetectors.jl#382] [Q&A] [answered]
Dear Dr. Hagemann 

I've been studying charge sharing of my detector using SolidStateDetectors (SSD) and I have some questions about chargecloud models of SSD. 

SSD considers charge clouds through function `NBodyChargeCloud`  which distributes a given energy deposition at a position to a central charge surrounded by shells of approximately N point charges, and then drifts the charges to calculate the signal. This approach seems to be an approximation in order to save simulation time. However, theoretically, the induced signal  would require drift each carrier (electron or hole) individually rather than a “charge” in the charge cloud structure.
Regarding this, I have the following questions:

① I am curious about your views on this approximation. Have you evaluated the errors associated with drifting "charge" in a charge cloud structure compared to drifting individual carriers?  It seems that the error depend on the settings of the `number of charges` and `number_of_shells`?

②parameters` number of charges` and `number_of_shells` slow down the simulation if set higher. Besides this, are there other standards that can guide or validate how to set their values? Specifically, number_of_shells affects the entire size of the charge cloud. I would appreciate any advice for determining them.

③The radius of the charge cloud is constant, Beta and Gamma is 0.5mm. 
```
radius_guess(charge::T, ::Type{Beta}) where {T} = T(0.0005)
radius_guess(charge::T, ::Type{Gamma}) where {T} = T(0.0005) 
```
How was this value determined? Could you provide any references?

④Since `radius_guess` returns constant, once `number of charges `and `number_of_shells `are determined, the size of the charge cloud remains the same for **different deposition energies**. Intuitively, I feel that the larger the energy, the larger the size of the charge cloud should be. Could you provide some insights of the case?

Thank you for your assistance and looking forward to your reply.
Best regards!

**Comment by fhagemann** [comment, marked as answer]
In general, SSD provides a simulation framework with which functions in the code can easily modified by users if needed.
As we state in [our documentation](https://juliaphysics.github.io/SolidStateDetectors.jl/main/man/charge_drift/#Group-Effects), charge clouds, diffusion and self-repulsion are implemented as experimental features. Charge clouds are modeled as groups of point-like objects, but we do not model each electron and hole individually (especially because the run time for charge drift simulations scales with the number of charge carriers and millions of charges would need to be simulated for a 3MeV event, for example).

Now to your questions:
1. You can evaluate the error e.g. for a 10keV event, where the expected amount of electrons and holes is still reasonably small enough to simulate. Compare the simulation output for the two cases (our charge cloud approach and individual electrons and holes), and see how big the deviations are between the two. From this, you can estimate the error expected when simulating events at higher energies.
2. If you perform the comparison as indicated in my first answer, then you can see how big the deviations are for different sets of `number_of_carries` and `number_of_shells` and choose values, for which the accuracy and the run time of the simulation suits you best.
3. These are just dummy values that can be overwritten by the user. You can redefine
    ```julia
     SolidStateDetectors.radius_guess(charge::T, ::Type{SolidStateDetectors.Beta}) where {T}  = T(0.001)
   ``` 
   to be any other value you like (in this case $1\text{mm}$ for interactions from beta particles).
4. `radius_guess` takes two arguments: `charge` and `ParticleType`. You can overwrite the function to use the charge value when calculating the radius. 
    ```julia
    SolidStateDetectors.radius_guess(charge::T, Type{SolidStateDetectors.Beta}) where {T} = T(0.0005 + 1e-10 * charge)
    ```
    Note that the charge is given in units of `C` (SSD's internal energy unit) and that the result should give the radius in `m` (SSD's internal length unit). This example would be $r(Q) = 0.5\text{mm} + 10^{-7}\frac{\text{mm}}{\text{C}} \cdot Q$.

**Reply by zql2021** [reply]
Thank you for your detailed answer! I will have a try as you suggested.
